pkg/sql/logictest/testdata/logic_test/sequences

```diff
@@ -0,0 +1,180 @@
+
+statement error pq: unimplemented at or near "EOF"
+CREATE SEQUENCE err_test OWNED BY someuser
+
+# DML & DDL ERRORS
+
+statement error pgcode 42809 cannot run INSERT on sequence "foo" - sequences are not updateable
+INSERT INTO foo VALUES (1, 2, 3)
+
+statement error pgcode 42809 cannot run UPDATE on sequence "foo" - sequences are not updateable
+UPDATE foo SET value = 5
+
+statement error pgcode 42809 cannot run DELETE on sequence "foo" - sequences are not updateable
+DELETE FROM foo
+
+statement error pgcode 42809 cannot run TRUNCATE on sequence "foo" - sequences are not updateable
+TRUNCATE foo
+
+# Drop table on sequences doesn't work; you have to use DROP SEQUENCE.
+statement error pgcode 42809 "foo" is not a table
+DROP TABLE foo
+
+# USING THE nextval() FUNCTION
+
+# `select * from foo` currently returns 0 columns & 0 rows;
+# testing against that is a Test-Script syntax error.
+
+query I
+SELECT nextval('foo')
+----
+1
+
+query I
+SELECT nextval('foo')
+----
+2
+
+# You can create a sequence with different increment.
+
+statement ok
+CREATE SEQUENCE bar INCREMENT 5
+
+query I
+SELECT nextval('bar')
+----
+1
+
+query I
+SELECT nextval('bar')
+----
+6
+
+# You can create a sequence with different start and increment.
+
+statement ok
+CREATE SEQUENCE baz START 2 INCREMENT 5
+
+query I
+SELECT nextval('baz')
+----
+2
+
+query I
+SELECT nextval('baz')
+----
+7
+
+# You can create and use a sequence with special characters.
+
+statement ok
+CREATE SEQUENCE spécial
+
+query I
+SELECT nextval('spécial')
+----
+1
+
+# You can't call nextval on a table.
+
+statement ok
+CREATE TABLE kv (k bytes primary key, v bytes)
+
+statement error pgcode 42809 "kv" is not a sequence
+SELECT nextval('kv')
+
+# Parse errors in the argument to nextval are handled.
+statement error pq: nextval\(\): syntax error at or near "@"
+SELECT nextval('@#%@!324234')
+
+# You can create and find sequences from other databases.
+
+statement ok
+CREATE DATABASE other_db
+
+statement ok
+SET DATABASE = other_db
+
+statement ok
+CREATE SEQUENCE other_db_test
+
+statement ok
+SET DATABASE = test
+
+# Sequence names are resolved based on the search path.
+
+statement ok
+CREATE DATABASE foo
+
+statement ok
+CREATE DATABASE bar
+
+statement ok
+CREATE SEQUENCE foo.x
+
+statement ok
+SET DATABASE = bar
+
+statement ok
+SET SEARCH_PATH = foo
+
+query I
+SELECT nextval('x')
+----
+1
+
+query I
+SELECT nextval('other_db.other_db_test')
+----
+1
+
+# USE WITH TABLES
+
+# You can use a sequence in a DEFAULT expression to create an auto-incrementing primary key.
+
+statement ok
+CREATE SEQUENCE blog_posts_id_seq
+
+statement ok
+CREATE TABLE blog_posts (id INT PRIMARY KEY DEFAULT nextval('blog_posts_id_seq'), title text)
+
+statement ok
+INSERT INTO blog_posts (title) values ('foo')
+
+statement ok
+INSERT INTO blog_posts (title) values ('bar')
+
+# TODO(vilterp) change this to [1, 2] once initial-evaluation bug has been fixed
+query I
+SELECT id FROM blog_posts ORDER BY id
+----
+2
+3
```

Comment:
Curious for some advice from you, @knz:

In this example, we create a new sequence and then use it in a DEFAULT expression. However, when we insert the first row into the table, the primary key comes out as 2, not 1 (where the sequence is supposed to start). This is because default expressions are run once at table creation time to see if they work (https://github.com/cockroachdb/cockroach/blob/d73de92d4e4f1e4b3c85e4c13b14ff6474391d6a/pkg/sql/sqlbase/table.go#L198-L205).

Would it be reasonable to flag functions which have side effects (e.g. writing to the database, as in this case), and exclude them from being run here? (Note also that we can't run it in a transaction and roll it back, because the `nextval` function operates in its own transaction)

If the solution is introduce a new flag on `Builtin` and use it to mark `nextval` as side effectful, maybe using the same terminology as postgres's volatility categories (https://www.postgresql.org/docs/8.2/static/xfunc-volatility.html) for these flags would be appropriate.

Also FWIW, I think this could be fixed in another PR if all else is good to go.

Reply:
So we do already have an `impure` flag on builtins and it does already propagate to those volatility categories, FWIW.